libs/pricing_model/src/lib.rs

```diff
@@ -0,0 +1,338 @@
+use std::borrow::Cow;
+use std::collections::HashMap;
+
+/// Maintains latest [`Segment`] for each of the branches, faciliates the pricing model
+/// calculations.
+pub struct Storage<K: 'static> {
+    segments: Vec<Segment>,
+    branches: HashMap<K, usize>,
+}
+
+/// State of a branch.
+#[derive(Clone, Debug, Eq, PartialEq)]
+pub struct Segment {
+    parent: Option<usize>,
+
+    /// Description of how did we get to this state.
+    pub op: Cow<'static, str>,
```

Comment:
Not sure we can really get that information from a live system

Reply:
Adding a comment on what methods use this, and using `const NO_OP = Cow::Borrowed("");` in `scenario_6` in an attempt to clarify this.